services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers.py

```diff
@@ -0,0 +1,241 @@
+    responses={
+        status.HTTP_500_INTERNAL_SERVER_ERROR: {"description": "Errors in container"}
+    },
+)
+async def containers_docker_inspect(
+    only_status: bool = Query(
+        False, description="if True only show the status of the container"
+    ),
+    shared_store: SharedStore = Depends(get_shared_store),
+) -> Dict[str, Any]:
+    """
+    Returns entire docker inspect data, if only_state is True,
+    the status of the containers is returned
+    """
+
+    def _format_result(container_inspect: Dict[str, Any]) -> Dict[str, Any]:
+        if only_status:
+            container_state = container_inspect.get("State", {})
+
+            # pending is another fake state use to share more information with the frontend
+            return {
+                "Status": container_state.get("Status", "pending"),
+                "Error": container_state.get("Error", ""),
+            }
+
+        return container_inspect
+
+    with docker_client() as docker:
+        container_names = shared_store.container_names
+
+        results = {}
+
+        for container in container_names:
+            try:
+                container_instance = await docker.containers.get(container)
+                container_inspect = await container_instance.show()
+                results[container] = _format_result(container_inspect)
+            except aiodocker.exceptions.DockerError as err:
+                _raise_from_docker_error(err)
+
+        return results
+
+
+@containers_router.get(
+    "/containers/{id}/logs",
+    responses={status.HTTP_404_NOT_FOUND: {"description": "Container does not exists"}},
+)
+async def get_container_logs(
+    id: str,
+    since: int = Query(
+        0,
+        title="Timestamp",
+        description="Only return logs since this time, as a UNIX timestamp",
+    ),
+    until: int = Query(
+        0,
+        title="Timestamp",
+        description="Only return logs before this time, as a UNIX timestamp",
+    ),
+    timestamps: bool = Query(
+        False,
+        title="Display timestamps",
+        description="Enabling this parameter will include timestamps in logs",
+    ),
+    shared_store: SharedStore = Depends(get_shared_store),
+) -> Union[str, Dict[str, Any]]:
+    """Returns the logs of a given container if found"""
+    # TODO: remove from here and dump directly into the logs of this service
+    # do this in PR#1887
+    _raise_if_container_is_missing(id, shared_store.container_names)
+
+    with docker_client() as docker:
+        try:
+            container_instance = await docker.containers.get(id)
+
+            args = dict(stdout=True, stderr=True, since=since, until=until)
+            if timestamps:
+                args["timestamps"] = True
+
+            container_logs: str = await container_instance.log(**args)
+            return container_logs
+        except aiodocker.exceptions.DockerError as err:
+            _raise_from_docker_error(err)
+            return None
```

Comment:
this return is probably enforced by the linter because he does not know about ``_raise_...``` .
I know it is a matter of style but I would add the raise directly here instead of using a helper function. But this is just an opinion on style :-)

Reply:
I have the same issue, but in the end the linter is annoying. It would be great if I could raise the function raising the error 🤦

Reply:
An alternative might be to use [try/except/ELSE/finally](https://docs.python.org/3/tutorial/errors.html#handling-exceptions)
```python
try:
  container_logs = await foo()
except ...
     _raise_...
else:
     return container_logs
finally:
     ... 
```

Reply:
to be honnest I find very weird this return after a function _raise... can the linter not be tuned a bit?

Reply:
I will make sure that the contextmanager which handles the docker connection will also trap log and raise exceptions.